mmr-lib/Randomizer.cs

```diff
@@ -445,7 +455,27 @@ private void ProcessDependenciesForItem(ItemObject currentItem, string line)
 
         public void SeedRNG()
         {
-            Random = new Random(_settings.Seed);
+            //generate the "public" seed
+            string seed = _settings.Seed.Trim();
+            if (string.IsNullOrWhiteSpace(seed))
+            {
+                seed = Environment.TickCount.ToString();
+            }
+            _settings.Seed = seed;
+
+            string field = _settings.GetGenerationSettings();
+            if(!_settings.OutputSpoiler)
+            {
+                field += "bad salt by mzxrules";
```

Comment:
I don't think this is necessary. I don't see why presence or absence of spoiler log should change the seed. Plus you can just output a spoiler log and add `bad salt by mzxrules` to your seed to get the same result, right?

Reply:
Races. Also, that's why it's a bad salt lol.

Reply:
Race participants shouldn't even be given the seed. They should just be given the patch. If you have the seed, there's no reason you shouldn't be able to generate the spoiler log for it.

Reply:
adding `bad salt by mzxrules` to the seed won't generate the same result, since the user input seed is no longer the same seed that the program uses to generate a shuffle. This is similar to how OoTR implements it.

Reply:
Oh I see. So the seed is just the hash of the generation settings json, and the generation settings includes the seed. So if all the generation settings are part of the seed, does that mean you can't for example generate two identical seeds one with random BGM and one without?
Also i still don't see the point of the salt though.

Reply:
BGM isn't part of the GenerationSettings, so it doesn't (or shouldn't) affect the internal seed